File: CPP/linearSearch.cpp
// Linear Search
// Works for integer arrays
// Returns the index of the first found element. -1 if not found

#include <iostream>

using namespace std;

template <size_t N>
int linearSearch(int (&arr)[N], int value);

int main(){
    int myArray[] = {2, 3, 7, 7, 11, 15, 25};
    int number = 7;
    int index = linearSearch(myArray, number);
    
    cout << "Index of " << number << " = " << index;
    return 0;
}

template <size_t N>
int linearSearch(int (&arr)[N], int value) {
    for (int i=0; i < N; ++i) {
        if (arr[i] == value) { // If the element is found
            return i; // Return the index of the element
        }
    }

    return -1; // Element not found
}

File: Python/linear_search.py
# Tested with Python 3.13.2

import dis

def linear_search(array, value):
    '''
    Return the index of given value in the givenn array.
    
    Parameters:
        array (list): A list of values.
        value (int): The value to search for.
        
    Returns:
        int: The index of the value in the array.
    '''
    for i, crnt_val in enumerate(array):
        if crnt_val == value:
            return i
    
    return -1

# EXAMPLE USAGE

arr = [3, 7, 2, 9, 5]
targetVal = 9

result = linear_search(arr, targetVal)

if result != -1:
    print("Value",targetVal,"found at index",result)
else:
    print("Value",targetVal,"not found")
    

dis.dis(linear_search)